app/models/concerns/executable.rb

```diff
@@ -0,0 +1,44 @@
+# Abstract class for defining a standardized CLI model
+class Executable
+  require 'cheetah'
+  require 'json'
+
+  class CliError < StandardError; end
+
+  include ActiveModel::Model
+
+  attr_accessor :region, :tag_scope
+
+  def self.load()
+    new(
+      region: KeyValue.get(:region),
+      tag_scope: KeyValue.get('tag_scope', 'suse-rancher-setup')
+    )
+  end
+
+  def environment()
+    raise NotImplementedError
+  end
+
+  def command()
+    raise NotImplementedError
+  end
+
+  def execute(*args)
+    stdout, stderr = Cheetah.run(
+      [self.command, *args.flatten],
+      stdout: :capture,
+      stderr: :capture,
+      env: self.environment,
+      logger: Logger.new(Rails.configuration.cli_log)
+    )
+    return stdout
```

Comment:
missing `raise StandardError.new(stderr) if stderr.present?` 
We missed errors because of we did not have that command

Reply:
which should be 
```ruby
raise CliError.new(stderr) if stderr.present?
```

Reply:
:thinking:
If the return code is anything but `0`, `Cheetah::ExecutionFailed` is raised. If return code _is_ `0`, then the output we want is in `stdout`. I'm not aware of a case where the retcode is 0, but stderr contains valid results.

Reply:
We had some cases with AWS testing where that happened
a command that exited with return code 0 but had `stderr` populated 
and it did not work/failed

also 
```
If the execution succeeds, the returned value depends on the value of the
 `:stdout` and `:stderr` options (see below). If the execution fails, the
 method raises an {ExecutionFailed} exception with detailed information
 about the failure. (In the single command case, the execution succeeds if
 the command can be executed and returns a zero exit status. In the
 multiple command case, the execution succeeds if the last command can be
 executed and returns a zero exit status.
 ```
 from [cheetah repo](https://github.com/openSUSE/cheetah/blob/master/lib/cheetah.rb#L251)

Reply:
Added in 8cf629a.